htdocs/user/home.php

```diff
@@ -102,21 +102,14 @@
 $sql.= " FROM ".MAIN_DB_PREFIX."user as u";
 $sql.= " LEFT JOIN ".MAIN_DB_PREFIX."societe as s ON u.fk_soc = s.rowid";
 // TODO add hook
-if (! empty($conf->multicompany->enabled)) {
-	if (! empty($conf->global->MULTICOMPANY_TRANSVERSE_MODE)) {
-		if (! empty($user->admin) && empty($user->entity)) {
-			if ($conf->entity == 1) {
-				$sql.= " WHERE u.entity IS NOT NULL";
-			} else {
-				$sql.= " WHERE u.entity IN (".getEntity('user').")";
-			}
-		} else {
-			$sql.= ",".MAIN_DB_PREFIX."usergroup_user as ug";
-			$sql.= " WHERE ug.fk_user = u.rowid";
-			$sql.= " AND ug.entity IN (".getEntity('user').")";
-		}
+if (! empty($conf->multicompany->enabled) && ! empty($conf->global->MULTICOMPANY_TRANSVERSE_MODE)) {
+	if (! empty($user->admin) && empty($user->entity) && $conf->entity == 1) {
+		$sql.= " WHERE u.entity IS NOT NULL"; // Show all users
 	} else {
-		$sql.= " WHERE u.entity IN (".getEntity('user').")";
+		$sql.= ",".MAIN_DB_PREFIX."usergroup_user as ug";
```

Comment:
If i am an "admin" user on entity 4 and no group was yet defined/assigned to user, listing the users was ok with old code, but with the new one, the list will always be empty (because of join on usergroup_user). It means the admin user of entity 4 will see nobody. Old code was correctly managing this case. 
2 solution:
- May be a left join on usergroup_user may be the solution.
- Or just add the " OR u.entity = 0" at end of line 111 of old code

Reply:
@eldy 
in "transverse mode" is mandatory to link "user + group + entity", admin too, except "superadmin"

Reply:
Even, if you associate the admin user with a group and entity, when this user read list of user, he can't see users that has no groups yet in entity 4 with this change, so he can't make the link to associate not linked users because the users are not reachable/visible until the link is done.
This change make visibility of database not consistent with its content and no way to fix this except by making direct update into database.

Why not just adding the " OR u.entity = 0" at end of line 111 of old code
This will provide the feature you need, and does not bring to deadlock situation.

Reply:
@eldy 
in transverse mode an admin just view users associated with an entity !
just "superadmin" can associated a user with an entity.